### PR TITLE
TFAC-112 Adding LogUserRevenueMutation

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: src/__tests__/pages/index.test.js [hunk 1]
```diff
@@ -249,4 +278,159 @@ describe('index.js', () => {
     mount(<IndexPage {...mockProps} />)
     expect(LogTabMutation).not.toHaveBeenCalled()
   })
+
+  it('calls LogUserRevenueMutation for each Ad when the onAdDisplayed prop is invoked', () => {
+    setUpAds()
+    const IndexPage = require('src/pages/index').default
+    const wrapper = mount(<IndexPage {...getMockProps()} />)
+    /* eslint-disable no-console */
+    console.log(wrapper.debug())
```
Comment (kmjennison) [inline src/__tests__/pages/index.test.js:287]: Remove console log

File: src/__tests__/pages/index.test.js [hunk 2]
```diff
@@ -249,4 +278,159 @@ describe('index.js', () => {
...
+      adUnitCode: '/12345/ThirdAdHere',
+    })
+  })
+
+  it('does not call LogUserRevenueMutation when the ad info is null', () => {
+    setUpAds()
+    const IndexPage = require('src/pages/index').default
+    const wrapper = mount(<IndexPage {...getMockProps()} />)
+    /* eslint-disable no-console */
+    console.log(wrapper.debug())
```
Comment (kmjennison) [inline src/__tests__/pages/index.test.js:369]: Remove

File: src/__tests__/pages/index.test.js [hunk 3]
```diff
@@ -249,4 +278,159 @@ describe('index.js', () => {
...
+      adUnitCode: '/12345/SomeAdUnit',
+    })
+  })
+
+  it('does not include encodedRevenue in the LogUserRevenueMutation when the encodedRevenue value is nil', () => {
+    setUpAds()
+    const IndexPage = require('src/pages/index').default
+    const wrapper = mount(<IndexPage {...getMockProps()} />)
+    /* eslint-disable no-console */
+    console.log(wrapper.debug())
```
Comment (kmjennison) [inline src/__tests__/pages/index.test.js:408]: Remove

File: src/pages/index.js [hunk 4]
```diff
@@ -312,6 +314,35 @@ const Index = ({ data: initialData }) => {
...
+      adSize,
+      // Only send aggregationOperation value if we have more than one
+      // revenue value
+      aggregationOperation:
+        !isNil(revenue) && !isNil(encodedRevenue) ? 'MAX' : null,
+      tabId: context.tabId,
+      adUnitCode: GAMAdUnitId,
+    })
+
     console.log('Ad displayed:', displayedAdInfo, context) // eslint-disable-line no-console
```
Comment (kmjennison) [inline src/pages/index.js:346]: Go ahead and remove this console log, too.

File: src/utils/mutations/__tests__/LogUserRevenueMutation.test.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* eslint-env jest */
```
Comment (kmjennison) [inline src/utils/mutations/__tests__/LogUserRevenueMutation.test.js:1]: You can remove this eslint line. Our config in this repo will assume `*.test.js` files are a Jest env.

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: src/__tests__/pages/index.test.js (hunk 1 above)
Comment (jedtan) [reply src/__tests__/pages/index.test.js:287]: Whoops!

**Pull request merged**